### PR TITLE
Helm tests and release

### PR DESCRIPTION
**Pull request by pdk27**
*Issue #, if available:*
N/A

*Description of changes:*
* improve update-versions-for-release script
* add release version validator
* move validators under scripts
* modify Makefile to accommodate above changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 VERSION ?= $(shell git describe --tags --always --dirty)
 LATEST_TAG=$(shell git tag | tail -1)
+SECOND_LATEST_TAG=$(shell git tag | tail -2 | head -1)
```
Comment (brycahta) [inline Makefile:3]: 'previous' would be a clearer name than 'second'

File: scripts/update-versions-for-release [hunk 2]
```diff
@@ -56,11 +57,11 @@ update_versions_in_chart() {
 
     sed -i '' "s/version: $old_chart_version/version: $new_chart_version/g" $CHART
     sed -i '' "s/amazon-ec2-metadata-mock-$old_chart_version.tgz/amazon-ec2-metadata-mock-$new_chart_version.tgz/g" $CHART_README
-    echo -e "\nUpdated chart version from $old_chart_version to $new_chart_version in files: \n$CHART\n$CHART_README"
+    echo -e "\n✅ Updated chart version from $old_chart_version to $new_chart_version in files: \n$CHART\n$CHART_README"
```
Comment (brycahta) [inline scripts/update-versions-for-release:60]: just fyi: In a PR I'm working on, I think we should remove the chart version from the **.tgz** and therefore the readme as well. Reasoning is the download link to the .tgz already contains the release version as an identifier for which charts a user downloads.

File: scripts/update-versions-for-release [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 
-# **Run this script and commit changes BEFORE creating a new release tag in Github**
+# **Run this script and commit changes AFTER creating a local tag for the new release and BEFORE creating a remote tag for the new release**
```
Comment (brycahta) [inline scripts/update-versions-for-release:3]: Just confirming: creating local tag means tagging a drafted release (via GitHub UI)?

File: Makefile [hunk 4]
```diff
@@ -36,25 +40,22 @@ compile:
 	go build -a -tags aemm${GOOS} -ldflags '-X "${DEFAULT_VALUES_VAR}=${ENCODED_METADATA_DEFAULTS}" -X "${ROOT_VERSION_VAR}=${VERSION}"' -o ${BUILD_DIR_PATH}/${BINARY_NAME} ${MAKEFILE_PATH}/cmd/amazon-ec2-metadata-mock.go
 
 validate-json:
-	${MAKEFILE_PATH}/test/json-validator
+	${MAKEFILE_PATH}/scripts/validators/json-validator
```
Comment (brycahta) [inline Makefile:43]: thanks for reorganizing 👍🏼

File: scripts/update-versions-for-release [hunk 5]
```diff
@@ -1,6 +1,7 @@
...
+# Pushing the local tag to remote / creating a new remote tag on Github will trigger the release process.
```
Comment (brycahta) [inline scripts/update-versions-for-release:4]: consider renaming `NEW_VERSION` and `OLD_VERSION` below to match with Latest/Previous

File: scripts/validators/release-version-validator [hunk 6]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Script to verify that AEMM version is updated to reflect the latest release in:
```
Comment (brycahta) [inline scripts/validators/release-version-validator:3]: Very clean! Is it possible to run this first, then have it trigger `update-versions-for-release` if diffs are detected? Or maybe edit Makefile with a target to do both at once.

I like the automation, but it also seems like it's more manual with understanding all the steps to take during a release to make sure this is all aligned

File: Makefile (hunk 1 above)
Comment (pdk27) [reply Makefile:3]: Yea! I thought about it but picked second latest:P 
I can change it to `previous`

File: scripts/update-versions-for-release (hunk 2 above)
Comment (pdk27) [reply scripts/update-versions-for-release:60]: Helm charts usually come with chart versions - I think this adds clarity, before, during and after downloading a chart archive. See [this](https://helm.sh/docs/topics/charts/#charts-and-versioning) for details. 

And updating the chart version is already automated, and is done along with other version updates, so shouldn't be problematic.

File: scripts/update-versions-for-release (hunk 3 above)
Comment (pdk27) [reply scripts/update-versions-for-release:3]: creating a local tag = `git tag v1.0.2`  -- this is how we can break the cyclic dependency in the release process (commit with release version updates <-> tag push). 
With local tag creation, we can run our scripts like `update-versions-for-release` and get that commit merged in before creating the remote tag  and triggering a Travis build for release.  

creating a remote tag - this is expected to be done on Github UI 

I think the steps will be clear once I document the release process.

File: scripts/update-versions-for-release (hunk 5 above)
Comment (pdk27) [reply scripts/update-versions-for-release:4]: good point 👍

File: scripts/validators/release-version-validator (hunk 6 above)
Comment (pdk27) [reply scripts/validators/release-version-validator:3]: There is definitely scope to automate the release prep further. 
For now, I wanted to keep it manual so we can test this workflow for a few releases, before making Travis commit to the repo.

File: Makefile [hunk 7]
```diff
@@ -1,5 +1,6 @@
 VERSION ?= $(shell git describe --tags --always --dirty)
 LATEST_TAG=$(shell git tag | tail -1)
+PREVIOUS_TAG=$(shell git tag | tail -2 | head -1)
```
Comment (brycahta) [inline Makefile:3]: Sorry, I meant `PREVIOUS_LATEST_TAG` . I should've been clearer 🤦🏻‍♂️

**Pull request merged**